tests/virt_autotest/lib/virt_utils.pm

```diff
@@ -0,0 +1,38 @@
+# SUSE's openQA tests
+#
+# Copyright © 2009-2013 Bernhard M. Wiedemann
+# Copyright © 2012-2016 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+package virt_utils;
+use strict;
+use warnings;
+use File::Basename;
+use base "opensusebasetest";
+use testapi;
+use Exporter;
+use Data::Dumper;
+use XML::Writer;
+use IO::File;
+
+our @EXPORT=qw(set_serialdev);
+
+sub set_serialdev(){
+    if (get_var("XEN") || check_var("HOST_HYPERVISOR","xen")){
+        my $hostReleaseInfo=script_output("cat /etc/SuSE-release");
+        if ($hostReleaseInfo =~ /VERSION = 12\nPATCHLEVEL = 2\n/m){
+            $serialdev = "hvc0";
+        }else {
+            $serialdev = "xvc0";
+        }
+    }else {
+        $serialdev = "ttyS1";
```

Comment:
This function will only be called inside our test scripts, which means we only *change* the SERIALDEV from initial value( $serialdev = get_var('SERIALDEV', "ttyS0")) when needed, and what it has to be changed to depends on the ouput of file /etc/SuSE-release, so we are not able to set  SERIALDEV correctly before we read the file.
Here is the usage in one typical test scenario: in host upgrade test from sles11sp4 to sles12sp2 xen, during installation, the default value ttyS1 is ok, then when boot to the sles11sp4 xen, we need to call this func to change it to xvc0, then after upgrade to sles12sp2 xen and boot into it, we will call this func again to set it to hvc0.

Do you agree?